src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsBaseObject.java

```diff
@@ -556,10 +557,25 @@ public AdsBaseObject verifyAdImage(String slotName, String src, String imageUrl)
 
   public AdsBaseObject verifyProvidersChain(String slotName, String providers) {
     PageObjectLogging.log("SlotName", slotName, true);
-    Assertion.assertEquals(providers, Joiner.on("; ").join(getProvidersChain(slotName)));
+    List<String> actualProviders = getProvidersChainWithWait(slotName, providers, PROVIDER_CHAIN_TIMEOUT_SEC);
+    Assertion.assertEquals(providers, Joiner.on("; ").join(actualProviders));
     return this;
   }
 
+  private List<String> getProvidersChainWithWait(final String slotName, final String expectedProviders, int timeoutSec) {
+    return new WebDriverWait(driver, timeoutSec).until(
+        new ExpectedCondition<List<String>>() {
+          @Override
+          public List<String> apply(WebDriver webDriver) {
+            if (expectedProviders.equals(Joiner.on("; ").join(getProvidersChain(slotName)))) {
+              return getProvidersChain(slotName);
+            }
+            return null;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Return an empty collection instead of null. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1168)